### ISSUE TITLE
Wrong Data on Energy Dashboard

### ISSUE THREAD

**Issue by jsagu** [wlcrs/huawei_solar#14]
Hi Guys

First many thanks for the amazing job. Im new to HA and just taken my first steps on it. 

I followed the readme file and I got my inversor connected and providing data but what I get looks really strange. 

![image](https://user-images.githubusercontent.com/97452525/148775572-12fa15ea-03dd-406b-9efa-4c3552a7c4dc.png)

![image](https://user-images.githubusercontent.com/97452525/148775617-e5781d55-955f-4d0d-b154-54ed0e37e4b9.png)

![image](https://user-images.githubusercontent.com/97452525/148775671-07517d28-29ab-4410-ba1b-5eba3bb7d167.png)

I was starting to play around with the data but before wanted to ensure that I did not put anything wrong.  BTW Inversor is a Huawei SUN2000 5KTL

**Comment by wlcrs** [comment]
I've identified the cause this weekend and was trialing the fix.

With the previous-to-last update, it was possible that HA was bombarding the inverter with update requests, causing it to return 0 instead of the proper value. This caused HA to think that the value had reset, leading to "flickering" between 0 and the proper daily value. HA sums the values between each reset, leading to the unrealistic high values.

I've extracted the fix from all my diagnostic code that I created to find the issue and pushed it to the repo.  Untested for the moment, but hopefully it works for you?

![image](https://user-images.githubusercontent.com/2150060/148789134-d0ca9692-56ff-4c44-8d1b-e5eb65653cd0.png)
